src/Broadway/CommandHandling/ClosureCommandHandler.php

```diff
@@ -31,13 +32,26 @@ public function add(\Closure $handler): void
         $reflection = new \ReflectionFunction($handler);
         $reflectionParams = $reflection->getParameters();
 
-        if (!isset($reflectionParams[0]) || !$reflectionParams[0]->getClass()) {
+        if (0 === $reflection->getNumberOfParameters()) {
             throw new ClosureParameterNotAnObjectException();
         }
 
-        $index = $reflectionParams[0]->getClass()->getName();
+        $reflectionType = $reflectionParams[0]->getType();
 
-        $this->handlers[$index] = $handler;
+        $name = null;
+        if ($reflectionType instanceof \ReflectionNamedType) {
+            /** @var \ReflectionNamedType $reflectionNamed */
+            $reflectionNamed = $reflectionParams[0]->getType();
+            if (!$reflectionNamed->isBuiltin()) {
+                $name = new ReflectionClass($reflectionNamed->getName());
+            }
+        }
+
+        if (!$name) {
+            throw new ClosureParameterNotAnObjectException();
+        }
+
+        $this->handlers[$name->getName()] = $handler;
```

Comment:
can we simplify it to something like this?

```
$reflection = new \ReflectionFunction($handler);
if (0 === $reflection->getNumberOfParameters()) {
    throw new ClosureParameterNotAnObjectException();
}

$reflectionType = $reflection->getParameters()[0]->getType();
if ($reflectionType instanceof \ReflectionNamedType && !$reflectionType->isBuiltin()) {
    $this->handlers[$reflectionType->getName()] = $handler;
} else {
    throw new ClosureParameterNotAnObjectException();
}
```

Reply:
I was thinking about this solution, usually, I don't like to use `else`  condition but sometimes creates a more clear code. We can think to add it now or trying to avoid  `else` but changing with something like this:

```
        if (!$reflectionType instanceof \ReflectionNamedType || $reflectionNamed->isBuiltin()) {
            throw new ClosureParameterNotAnObjectException();
        }

        $name = new ReflectionClass($reflectionParams[0]->getType()->getName());
```
Again, I don't like to use negative conditions, it's always a trade-off between best practice and readable code. And usually, I prefer readable code, but it's another solution.

In my PR I have splitter `isBuiltIn` check I think for some check failed